device/cuda/src/seeding/seed_finding.cu

```diff
@@ -347,7 +333,6 @@ seed_finding::output_type seed_finding::operator()(
                                       triplet_counter_midBot_buffer,
                                       triplet_buffer, seed_buffer);
     CUDA_ERROR_CHECK(cudaGetLastError());
-    m_stream.synchronize();
```

Comment:
The same here. The user of the algorithm will need to take care of not deleting the received buffer until the kernel filling it would've finished.